### PR TITLE
feat(settings-environments): danger zone

### PR DESCRIPTION
**Pull request by RemiBonnet**
# What does this PR do?

[> Link to the JIRA ticket
](https://qovery.atlassian.net/jira/software/projects/FRT/boards/23?selectedIssue=FRT-385)

- Add component global delete
Capture d’écran 2022-07-19 à 15 25 34

---

### Global

- [x] This PR does not introduce any breaking change
- [ ] This PR introduces breaking change(s) and has been labeled as such
- [x] I have found someone to review this PR and pinged him

### Store

- [ ] This PR introduces new store changes

### NX

- [x] I have run the dep-graph locally and made sure the tree was clean i.e no circular dependencies
- [x] I have followed the library pattern i.e `feature`, `ui`, `data`, `utils`

### Clean Code

- [x] I made sure the code is type safe (no any)
- [ ] I have included a feature flag on my feature, if applicable

### CODE REVIEW

**Comment by evoxmusic** [comment]
A preview environment was automatically created via Qovery.  
Click on the link below to follow its deployment and use it.  
:point_right: [[PR] staging - feat(settings-environments): danger zone - 2022-07-19T13:26:08Z](https://console.qovery.com/platform/organization/141c07c8-0dd9-4623-983b-3fdd61867255/projects/977bd069-20f7-4b24-bd95-28b44174cf43/environments/4966cc64-99e3-4a84-a34b-2cf5e9ae9983/applications)

File: libs/shared/ui/src/lib/components/block-content-delete/block-content-delete.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      <div className="p-5">
+        <p className="mb-5 text-sm text-text-600">{description}</p>
+        {list?.map((element, index) => (
+          <p key={index} data-testid={element.text} className="text-text-600 font-medium text-sm mb-2">
+            <Icon name={element.icon || 'icon-solid-trash'} className="mr-3 text-error-500" />
+            {element.text}
+          </p>
+        ))}
+        <div className="flex justify-end">
+          <Button className="mt-3 ml-auto" onClick={callback} style={ButtonStyle.ERROR}>
```
Comment (bdebon) [inline libs/shared/ui/src/lib/components/block-content-delete/block-content-delete.tsx:40]: There is no confirm dialog that open before doing the callback action? I thought that the confirm/popup part would be part of this component

File: libs/shared/ui/src/lib/components/block-content-delete/block-content-delete.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+    render(<BlockContentDelete {...props} />)
+
+    const firstElement = screen.getByTestId(props.list[0].text)
+    const secondElement = screen.getByTestId(props.list[1].text)
+
+    expect(firstElement.querySelector('.icon-solid-trash')?.classList.contains('icon-solid-trash')).toBe(true)
+    expect(secondElement.querySelector('.icon-solid-chart-area')?.classList.contains('icon-solid-chart-area')).toBe(
+      true
+    )
```
Comment (bdebon) [inline libs/shared/ui/src/lib/components/block-content-delete/block-content-delete.spec.tsx:35]: Need to add a test that simulate the click on the button to see if the callback has been called. 
act() click spy.hasbeencalled

File: libs/shared/ui/src/lib/components/block-content-delete/block-content-delete.tsx (hunk 1 above)
Comment (bdebon) [reply libs/shared/ui/src/lib/components/block-content-delete/block-content-delete.tsx:40]: I think for something that critical we need a confirm

**Review by bdebon — CHANGES_REQUESTED** [review]
Hey Remi,
I added some comments and suggestions directly to the code.

Also I have a question that has been in my mind lately. We have many components in the shared UI lib that do not have stories. It's most of them complex component like here with this `BlockContentDelete` 
It could be cool to have stories for these more complex components so that coming devs can know that they exist and that they can use them. Maybe we can add a subchapter in storybook named something like : `Composed Components` that are components made with an assembly of smaller components.

Open to discussion

File: libs/pages/services/src/lib/ui/page-settings-danger-zone/page-settings-danger-zone.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+export function PageSettingsDangerZone(props: PageSettingsDangerZoneProps) {
+  const { deleteEnvironment } = props
+
+  return (
+    <div className="flex flex-col justify-between w-full">
+      <div className="p-8">
+        <BlockContentDelete
+          className="w-[580px]"
```
Comment (bdebon) [inline libs/pages/services/src/lib/ui/page-settings-danger-zone/page-settings-danger-zone.tsx:14]: Can we put this value in the theme? 
If one day we want to increase the size of all our boxes we could in one go with the tailwind theme

**Pull request merged**